src/Views/Dock.vala

```diff
@@ -27,51 +27,38 @@ public class Dock : Gtk.Grid {
     construct {
         column_spacing = 12;
         halign = Gtk.Align.CENTER;
-        row_spacing = 6;
+        row_spacing = 12;
         margin_start = margin_end = 6;
 
-        var icon_size = new Granite.Widgets.ModeButton ();
-        icon_size.append_text (_("Small"));
-        icon_size.append_text (_("Normal"));
-        icon_size.append_text (_("Large"));
+        weak Gtk.IconTheme default_theme = Gtk.IconTheme.get_default ();
+        default_theme.add_resource_path ("/io/elementary/switchboard/plug/pantheon-shell");
 
-        Plank.Paths.initialize ("plank", Constants.PLANKDATADIR);
-        dock_preferences = new Plank.DockPreferences ("dock1");
+        var icon_size_32 = new Gtk.RadioButton (null);
+        icon_size_32.image = new Gtk.Image.from_icon_name ("application-default-icon-symbolic", Gtk.IconSize.DND);
+        icon_size_32.tooltip_text = _("Small");
 
-        var current = dock_preferences.IconSize;
+        var icon_size_48 = new Gtk.RadioButton.from_widget (icon_size_32);
+        icon_size_48.image = new Gtk.Image.from_icon_name ("application-default-icon-symbolic", Gtk.IconSize.DIALOG);
+        icon_size_48.tooltip_text = _("Medium");
```

Comment:
Do we want this to be "Medium", or "Normal" (as before)? Or probably better, "Default" like we do with text size?
```suggestion
        icon_size_48.tooltip_text = _("Default");
```